osu.Desktop.VisualTests/Tests/TestCaseBeatSyncedContainer.cs

```diff
@@ -0,0 +1,186 @@
+
+using osu.Framework.Testing;
+using osu.Framework.Graphics;
+using osu.Framework.Timing;
+using osu.Game.Overlays;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Graphics.Containers;
+using osu.Framework.Audio.Track;
+using osu.Game.Beatmaps.ControlPoints;
+using osu.Framework.Graphics.Shapes;
+using OpenTK.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Framework.Lists;
+using System;
+using System.Globalization;
+
+namespace osu.Desktop.VisualTests.Tests
+{
+    internal class TestCaseBeatSyncedContainer : TestCase
+    {
+        public override string Description => @"Tests beat synced containers.";
+
+        private MusicController mc;
+
+        public TestCaseBeatSyncedContainer()
+        {
+            Clock = new FramedClock();
+        }
+
+        public override void Reset()
+        {
+            base.Reset();
+            Clock.ProcessFrame();
+
+            Add(new BeatContainer
+            {
+                Anchor = Anchor.BottomCentre,
+                Origin = Anchor.BottomCentre,
+            });
+
+            Add(mc = new MusicController
+            {
+                Origin = Anchor.TopRight,
+                Anchor = Anchor.TopRight,
+            });
+            mc.State = Visibility.Visible;
+        }
+
+        private class BeatContainer : BeatSyncedContainer
+        {
+            private const int flash_layer_heigth = 150;
+
+            private readonly InfoString timingPointCount;
+            private readonly InfoString currentTimingPoint;
+            private readonly InfoString beatCount;
+            private readonly InfoString currentBeat;
+            private readonly InfoString beatsPerMinute;
+            private readonly InfoString beatLength;
+
+            private readonly Box flashLayer;
+
+            public BeatContainer()
+            {
+                RelativeSizeAxes = Axes.X;
+                AutoSizeAxes = Axes.Y;
+                Children = new Drawable[]
+                {
+                    new FillFlowContainer
+                    {
+                        Name = @"Info Layer",
+                        Anchor = Anchor.BottomLeft,
+                        Origin = Anchor.BottomLeft,
+                        Direction = FillDirection.Vertical,
+                        AutoSizeAxes = Axes.Both,
+                        Margin = new MarginPadding { Bottom = flash_layer_heigth, },
+                        Children = new Drawable[]
+                        {
+                            timingPointCount = new InfoString(@"Timing points amount: "),
+                            currentTimingPoint = new InfoString(@"Current timing point: "),
+                            beatCount = new InfoString(@"Beats amount (in the current timing point): "),
+                            currentBeat = new InfoString(@"Current beat: "),
+                            beatsPerMinute = new InfoString(@"BPM: "),
+                            beatLength = new InfoString(@"Beat length: "),

```

Comment:
call this "adjusted beat length" or something (and show non-adjusted).

also remove spaces from the end of these and add them in the class you made for it